### PR TITLE
Create a way to set up application debug context for each active job

### PR DESCRIPTION
**Pull request by sunil-sadasivan**
Resolves https://github.com/department-of-veterans-affairs/appeals-deployment/issues/125

### Description
This is the application side adjustment to allow us to tag each ActiveJob with an application associated with it.  This then allows us to route exceptions occurring in each active job to the right slack room and owner.

### CODE REVIEW

**Comment by sunil-sadasivan** [comment]
I _think_ I'm defining and setting an `application_attr` right.  Let me know if this looks off.

File: app/jobs/concerns/application_attributes.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Allows for a developer to specify which ActiveJob belongs to which application
+# This is used for proper debugging with sentry and other internal tools
+module ApplicationAttributes
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def application_attr(app_name)
```
Comment (shanear) [inline app/jobs/concerns/application_attributes.rb]: 1) This is simple and ubiquitous enough to where I think we can probably just put this directly in `ApplicationJob`
2) I think calling it `application` or `set_application` is nicer sounding 😄

File: app/jobs/application_job.rb [hunk 2]
```diff
@@ -1,2 +1,4 @@
 class ApplicationJob < ActiveJob::Base
+  include ApplicationAttributes
+  application_attr :intake
```
Comment (shanear) [inline app/jobs/application_job.rb]: should Job errors by default be intake errors? I don't think we need to call this in `ApplicationJob`

Comment (sansb) [inline app/jobs/application_job.rb]: why is intake the default?

Comment (sunil-sadasivan) [reply app/jobs/application_job.rb]: Ah oops, debugging 🙈

**Comment by shanear** [comment]
This still doesn't appear to be working for `SyncIntakeJob`

**Pull request merged**